### DISCUSSION TITLE
Problem with the Interactive segmentation using Points on Image

### DISCUSSION THREAD

**Discussion by JoelDJ2002** [opengeos/segment-geospatial#234] [Q&A] [answered]
I am facing a glitch in the Interactive segmentation using Points in image
<img width="365" alt="image" src="https://github.com/opengeos/segment-geospatial/assets/66190032/5abb8816-52fb-48c1-8ce4-89a258959fca">
When you select Foreground Point and then select Background points it automatically increments the number of background points. 
I restarted and reran the entire code 5 times. Apparantly I found out that it had choosen one points on its own outside the image somewhere randomly . 
It would be great if this issue could be resolved

**Comment by giswqs** [comment]
Thank you for reporting. I am looking into it.

**Comment by giswqs** [comment]
I can't reproduce the issue. When placing markers outside the image, it should throw an error. 

![Peek 2024-01-18 00-49](https://github.com/opengeos/segment-geospatial/assets/5016453/056dc853-e954-4135-a57c-d78bf80eee67)

**Comment by giswqs** [comment, marked as answer]
Points ouside the image boundary are now ignored. Fix in #240

**Reply by JoelDJ2002** [reply]
Thank you for replying . I was looking through the source code and then realised that as I change from foreground to background the count of the background increments to 1 . The point pin falls on the coordinates of the mouse cursor click . So it drops a pin ( or point for google colab ) to the position on the map where the widget radio buttons are on the map . I am uploading a screen cast. You can see as to how the counter increases. Same thing happens when I change it to foreground.

[Screencast from 19-01-24 11:00:27 AM IST.webm](https://github.com/opengeos/segment-geospatial/assets/66190032/62231462-f96d-4495-a365-faf4b75f801d)

**Reply by giswqs** [reply]
Yes, clicking on the make will drop a marker. That's the expected behavior. You should drag and drop the map if you just want to pan the map.

**Reply by JoelDJ2002** [reply]
But can we drag the points in colab ?
